app/src/main/java/com/realityexpander/tasky/presentation/register_screen/RegisterScreen.kt

```diff
@@ -1,14 +1,211 @@
 package com.realityexpander.tasky.presentation.register_screen
 
-import androidx.compose.material.Text
-import androidx.compose.runtime.Composable
+import androidx.activity.compose.BackHandler
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.platform.LocalSoftwareKeyboardController
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.unit.dp
 import androidx.hilt.navigation.compose.hiltViewModel
 import com.ramcosta.composedestinations.annotation.Destination
+import com.ramcosta.composedestinations.navigation.DestinationsNavigator
+import com.realityexpander.tasky.presentation.components.EmailField
+import com.realityexpander.tasky.presentation.components.PasswordField
+import com.realityexpander.tasky.presentation.destinations.LoginScreenDestination
+import kotlinx.coroutines.launch
 
+@OptIn(ExperimentalComposeUiApi::class)
 @Composable
-@Destination
+@Destination("RegisterScreen")
 fun RegisterScreen(
-    vieModel: RegisterViewModel = hiltViewModel()
+    email: String? = null,
+    password: String? = null,
+    confirmPassword: String? = null,
+    navigator: DestinationsNavigator,
+    viewModel: RegisterViewModel = hiltViewModel(),
 ) {
-    Text(text = "Register")
+    BackHandler(true) { /* We want to disable back clicks */ }
```

Comment:
Any reason for that?

Reply:
When I implement the splash screen, i don't want the user to go back to the splash screen. I haven't tried the splash screen processing yet, so I assumed that allowing the back button would return to the splash screen... maybe it doesn't?

And I have a button on the screen to flip back and forth between `Sign-In` and `Register`, so I also thought it kept the from returning to the previous screen.

2 questions:

1. So, should I allow the user to use the back button? It wont return to the splash screen, correct?
3. And when the user successfully logs in, should I clear the backstack for the `LoginScreen` and `RegistrationScreen`?

Reply:
I found a way to force only navigation between the login and registraion. If the user wants to quit, he will have to force quit the app currently